### DISCUSSION TITLE
How do I get Keycloak/Infinispan to tell me why it won't use my jgroup stack?

### DISCUSSION THREAD

**Discussion by jerrac** [keycloak/keycloak#28181] [Q&A]
Every time I try to adjust the jgroup configuration in cache-ispn.xml, I end up with messages like this:

```
ISPN000088: Unable to use any JGroups configuration mechanisms provided in properties {}. Using default JGroups configuration!
```

And then Keycloak just goes and sets up the default config, completely ignoring my jgroup.

There are NO logs surrounding that message indicating why it doesn't like my config.

I do have debug logs working. I can see plenty of them. I used `--log-level debug` on kc.sh.

I've tried about 50 different variations (no, not exaggerating, I've spent multiple hours over multiple days on this) on my config using the official [Infinispan docs](https://infinispan.org/docs/stable/titles/security/security.html#configuring-jgroups-sym-encrypt_secure-cluster-transport) and the config docs like: https://docs.jboss.org/infinispan/14.0/configdocs/infinispan-config-14.0.html as reference. Not to mention random web search results and other discussions and issues here on github.

I've found docs on [CLI](https://infinispan.org/docs/stable/titles/cli/cli.html) stuff for logging, and I think I saw something about log4j.xml. But I haven't found a way to access Infinispan's cli tool for the embedded Keycloak instance. And there's no sign of any log4j.xml files in the extracted Keycloak tar.gz.

So, how do I get Keycloak/Infinispan to tell me why it won't use my jgroup stack?

**Comment by codespearhead** [comment]
The `log4j.xml` configuration file is only available in the [WildFly distribution of Keycloak](https://www.keycloak.org/migration/migrating-to-quarkus), which is [over two years old by now](https://github.com/keycloak/keycloak/releases?q=17&expanded=true) and [it's not even an LTS release](https://github.com/keycloak/keycloak/discussions/26706#discussioncomment-9001438).

I suggest trying the [Configuring distributed caches](https://www.keycloak.org/server/caching) tutorial on the latest major version of Keycloak.

**Reply by jerrac** [reply]
Thanks for the reply, but if it's not a log4j thing, then how do you adjust Infinispan's log level?

FYI, I've been over all the Keycloak cache related documentation (including the page you linked) I can find many times. At this point if I haven't found it there by now, I'm not going to. If it is there, I'll need it pointed out directly.
